### ISSUE TITLE
Does Tensorflow install on 32-bit Windows 7 ? 

### ISSUE THREAD

**Issue by paulgureghian** [tensorflow/tensorflow#20362]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**:
- **Python version**: 
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
I could not install Tensorflow on my Win 7 32-bit machine through Anaconda or Pycharm, but I did
on my Win 7 64-bit machine.  

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by marshalhayes** [comment]
Hi @paulgureghian. TensorFlow is [only tested and supported for 64-bit, x86 systems](https://www.tensorflow.org/install/install_windows). I don't believe you can install TensorFlow through `pip` or `conda` normally from a 32-bit system. If you take a look [here](https://pypi.org/project/tensorflow/#files), only 64-bit systems are listed for Windows.

Perhaps you could try to [build and install TensorFlow from source instead](https://www.tensorflow.org/install/install_sources). But, to avoid the headache, maybe you could dual boot Linux :)

**Comment by paulgureghian** [comment]
Or just use Theano as the backend to Keras ?

**Comment by marshalhayes** [comment]
Actually, I think I may be wrong. I believe TensorFlow will work on your 32-bit system as long as you are using 64-bit Python. Can you try that? 

Besides what I've already mentioned, I don't think I can help much more. I'll let someone else pickup from here.

**Comment by paulgureghian** [comment]
how to run python 64 on a 32 bit system ?

**Comment by gunan** [comment]
You cannot run 64 bit software on a 32 bit OS.
Some users had success compiling TF on 32 bit systems, but we have not.
Unfortunately, that is all I have about this issue. You may try searching for relevant issues or stackoverflow questions to see what people did to build TF on 32 bit systems.

**Comment by marshalhayes** [comment]
You're right. Sorry to cause confusion. I'm not sure what I was thinking.

I hope you can get TensorFlow working for you, @paulgureghian

**Comment by fo40225** [comment]
Using docker on 64-bit Linux to make tensorflow worked is a better way.

If you insist, you can help test it.

https://github.com/fo40225/tensorflow-windows-wheel/tree/master/1.8.0/py36/CPU/sse2

**Comment by gunan** [comment]
On windows, you cannot have linux docker container. Also, on an 32 bit OS you cannot run 64 bit docker containers.
I think what you are proposing is this:
https://stackoverflow.com/questions/56124/can-i-run-a-64-bit-vmware-image-on-a-32-bit-machine

**Comment by fo40225** [comment]
@gunan You CAN run a linux docker container thought [docker for windows](https://docs.docker.com/docker-for-windows/#switch-between-windows-and-linux-containers), but it is based on vm and it didn't support gpu.

I have provided a 32 bit tensorflow built for windows, although it's not a helpful thing.

32-bit process can use only 2GB memory; a complex model would not load.

**Comment by tensorflowbutler** [comment]
As this issue has invited community support, please remove the assignee. Otherwise, remove the `community support` label. Thank you.

**Comment by VanitarNordic** [comment]
if you use Anaconda, you can use  `conda install -c hesi_m tensorflow ` to install **tensorflow 1.9.0** for CPU 

if you specifically use Keras also, you can use  `conda install -c hesi_m keras` which installs **Keras 2.2.2** in accompany with everything including tensorflow 1.9.0

It is better to make a fresh anaconda environment for your experiments

**Comment by arindam-mallik** [comment]
Hi 

I need to install and configure Tensorflow on my Windows-7 32-bit OS through Anaconda 5.3 Python 3.6 version and facing lot of issues. Please let me know steps to follow from scratch (and share if any specific installer I will start the process from scratch)

Thanks

**Comment by comann** [comment]
@arindam-mallik  Did you have any success building binaries for Windows 32-bit?

I am looking for built binaries .dll  and .lib for 32bit machines.

**Comment by paulgureghian** [comment]
Since building 32-bit binaries from the source code seemed like it might be a bit over my head, I just use TF on my 64-bit machine instead through Anaconda.

**Comment by arindam-mallik** [comment]
Tensorflow does not work for 32 bit os.

Also in 64 bit OS, if Nvidia graphics not available GPU version of
tensorflow we cannot but you can install CPU version.

Thanks & Regards
Arindam

**Comment by comann** [comment]
My use case is basic CPU operations only, though I do need 32 bit capabilities. I am looking into model conversion into another format that supports 32bit.

**Comment by paulgureghian** [comment]
What about Keras, Pytorch, Caffe, Theano, or MXNet ?

**Comment by arindam-mallik** [comment]
Keras will not work in 32 bit OS too.

I suggest you can utilize Google colaboratory lab and try your coding in
cloud.

Thanks
Arindam

**Comment by reliefs** [comment]
https://github.com/tensorflow/tensorflow/issues/32315

**Comment by mihaimaruseac** [comment]
Closing as #32315 duplicates it.

**Comment by reliefs** [comment]
https://github.com/tensorflow/tensorflow/issues/32627

**Comment by jahnog** [comment]
Hi, I have detailed instructions to compile and install Tensorflow in an 32 bits linux in this repository: [tensorflow-32-bits-linux](https://github.com/jahnog/tensorflow-32-bits-linux)

**Comment by Xx-ACEPREDATOR-xX** [comment]
> if you use Anaconda, you can use  `conda install -c hesi_m tensorflow ` to install **tensorflow 1.9.0** for CPU 
> 
> if you specifically use Keras also, you can use  `conda install -c hesi_m keras` which installs **Keras 2.2.2** in accompany with everything including tensorflow 1.9.0
> 
> It is better to make a fresh anaconda environment for your experiments

This command worked when installing tensorflow
